tests/test_pot_iterations.py

```diff
@@ -0,0 +1,216 @@
+            # required_iters too high
+            calculate_ip_iters(
+                test_constants.NUM_SPS_SUB_SLOT,
+                test_constants.NUM_SP_INTERVALS_EXTRA,
+                ssi,
+                sp_interval_iters,
+                sp_interval_iters,
+            )
+
+        with raises(ValueError):
+            # required_iters too high
+            calculate_ip_iters(
+                test_constants.NUM_SPS_SUB_SLOT,
+                test_constants.NUM_SP_INTERVALS_EXTRA,
+                ssi,
+                sp_interval_iters,
+                sp_interval_iters * 12,
+            )
+
+        with raises(ValueError):
+            # required_iters too low (0)
+            calculate_ip_iters(
+                test_constants.NUM_SPS_SUB_SLOT,
+                test_constants.NUM_SP_INTERVALS_EXTRA,
+                ssi,
+                sp_interval_iters,
+                uint64(0),
+            )
+
+        required_iters = sp_interval_iters - 1
+        ip_iters = calculate_ip_iters(
+            test_constants.NUM_SPS_SUB_SLOT,
+            test_constants.NUM_SP_INTERVALS_EXTRA,
+            ssi,
+            uint8(13),
+            required_iters,
+        )
+        assert (
+            ip_iters
+            == sp_iters
+            + test_constants.NUM_SP_INTERVALS_EXTRA * sp_interval_iters
+            + required_iters
+        )
+
+        required_iters = uint64(1)
+        ip_iters = calculate_ip_iters(
+            test_constants.NUM_SPS_SUB_SLOT,
+            test_constants.NUM_SP_INTERVALS_EXTRA,
+            ssi,
+            uint8(13),
+            required_iters,
+        )
+        assert (
+            ip_iters
+            == sp_iters
+            + test_constants.NUM_SP_INTERVALS_EXTRA * sp_interval_iters
+            + required_iters
+        )
+
+        required_iters = uint64(int(ssi * 4 / 300))
+        ip_iters = calculate_ip_iters(
+            test_constants.NUM_SPS_SUB_SLOT,
+            test_constants.NUM_SP_INTERVALS_EXTRA,
+            ssi,
+            uint8(13),
+            required_iters,
+        )
+        assert (
+            ip_iters
+            == sp_iters
+            + test_constants.NUM_SP_INTERVALS_EXTRA * sp_interval_iters
+            + required_iters
+        )
+        assert sp_iters < ip_iters
+
+        # Overflow
+        sp_iters = sp_interval_iters * (test_constants.NUM_SPS_SUB_SLOT - 1)
+        ip_iters = calculate_ip_iters(
+            test_constants.NUM_SPS_SUB_SLOT,
+            test_constants.NUM_SP_INTERVALS_EXTRA,
+            ssi,
+            uint8(test_constants.NUM_SPS_SUB_SLOT - 1),
+            required_iters,
+        )
+        assert (
+            ip_iters
+            == (
+                sp_iters
+                + test_constants.NUM_SP_INTERVALS_EXTRA * sp_interval_iters
+                + required_iters
+            )
+            % ssi
+        )
+        assert sp_iters > ip_iters
+
+    # def test_win_percentage(self):
```

Comment:
I think this commented-out code can be removed, right?

Reply:
Done